pym/portage/package/ebuild/doebuild.py

```diff
@@ -446,6 +447,11 @@ def doebuild_environment(myebuild, mydo, myroot=None, settings=None,
 	if eapi_exports_REPOSITORY(eapi) and "PORTAGE_REPO_NAME" in mysettings.configdict["pkg"]:
 		mysettings.configdict["pkg"]["REPOSITORY"] = mysettings.configdict["pkg"]["PORTAGE_REPO_NAME"]
 
+	if not eapi_exports_PORTDIR(eapi):
+		del mysettings["PORTDIR"]
+	if not eapi_exports_ECLASSDIR(eapi):
+		del mysettings["ECLASSDIR"]
```

Comment:
For filtering variables like this, I'd prefer that we use the `config.environ()` method. The `__delitem__` method is messy because it exhaustively deletes from all of the dictionaries in the config's lookuplist (separate dictionaries for make.conf, make.defaults, etc).

Reply:
Updated. Please disregard the first failure from travis, I have fixed that already.